### PR TITLE
scheme not being passed from packet into initial DKG

### PR DESCRIPTION
**Pull request by CluEleSsUK**

### CODE REVIEW

File: core/drand_beacon_test.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func TestMemDBBeaconJoinsNetworkAtStart(t *testing.T) {
 
 	// We want to explicitly run a node with the chain.MemDB backend
 	newNodes := ts.AddNodesWithOptions(t, 1, beaconName, WithDBStorageEngine(chain.MemDB))
-	group := ts.RunDKG()
+	group, _ := ts.RunDKG()
```
Comment (AnomalRoil) [inline core/drand_beacon_test.go:120]: Missing the require.NoError?

File: core/drand_beacon_test.go [hunk 2]
```diff
@@ -139,7 +139,7 @@ func TestMemDBBeaconJoinsNetworkAfterDKG(t *testing.T) {
 	beaconName := "default"
 
 	ts := NewDrandTestScenario(t, existingNodesCount, thr, period, beaconName)
-	group := ts.RunDKG()
+	group, _ := ts.RunDKG()
```
Comment (AnomalRoil) [inline core/drand_beacon_test.go:142]: Missing the require.NoError?

File: key/keys.go [hunk 3]
```diff
@@ -239,6 +239,8 @@ func (b ByKey) Less(i, j int) bool {
 	return bytes.Compare(is, js) < 0
 }
 
+var ErrInvalidKeyScheme = errors.New("could not marshall key - likely, the scheme did not match the beacon's scheme")
```
Comment (AnomalRoil) [inline key/keys.go:242]: could not **unmarshal** key, instead, no?

**Review by AnomalRoil — CHANGES_REQUESTED** [review]
Few changes. Also the demo code needs to be changed to match the new `InitDKGLeader` signature.

It might be good if @dlsniper could review the `RunDKG` changes since he's the one who added the channel-based errors, IIRC.

File: core/util_test.go [hunk 4]
```diff
@@ -299,23 +300,21 @@ func (d *DrandTestScenario) RunDKG() *key.Group {
 	defer cancel()
 	done := make(chan bool) // signal we are done with the reshare before the timeout
 
-	runLeaderNode := func() {
+	runLeaderNode := func() error {
```
Comment (dlsniper) [inline core/util_test.go:303]: If we make this change here, then the errors from `runLeaderNode` will not be passed as the function is launched with a `go runLeaderNode()`.

File: protobuf/drand/control.proto [hunk 5]
```diff
@@ -105,8 +105,8 @@ message InitDKGPacket {
     uint32 beacon_period = 3;
     // the minimum beacon period when in catchup.
     uint32 catchup_period = 4;
-    // indicates a set of values the process will use to act in specific ways
-    string schemeID = 5;
+    // since v1.5.0 this must be passed upon key generation
+    // string schemeID = 5;
```
Comment (dlsniper) [inline protobuf/drand/control.proto:109]: I think this should be marked as deprecated rather than fully removed.

**Review by dlsniper — CHANGES_REQUESTED** [review]

File: protobuf/drand/control.proto (hunk 5 above)
Comment (CluEleSsUK) [reply protobuf/drand/control.proto]: done

File: core/util_test.go (hunk 4 above)
Comment (CluEleSsUK) [reply core/util_test.go]: updated to use the channel

**Review by dlsniper — APPROVED** [review]
LGTM.

**Review by AnomalRoil — APPROVED** [review]
LGTM

**Pull request merged**